### PR TITLE
Adding Cmdlets to enable/disable custom domain SSL for frontdoor

### PR DESCRIPTION
**Pull request by pichandwork**
## Description

New cmdlets to enable HTTPS for custom domain in front door.
Enable-AzFrontDoorCustomDomainHttps
Disable-AzFrontDoorCustomDomainHttps
Get-AzFrontDoorFrontendEndpoint

Approved design review - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/256
## Checklist

- [ Yes] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ Yes] The title of the PR is clear and informative
- [ Yes] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [Yes] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [Yes ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [Yes ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [Yes ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by maddieclayton** [comment]
@azuresdkci Add to whitelist

**Comment by vladimir-shcherbakov** [comment]
@pichandwork  Please fix the Static Analysis issues: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/11257/artifact/artifacts/ValidateHelpIssues.csv/*view*/

**Comment by pichandwork** [comment]
@vladimir-shcherbakov - Fixed all issues. Can you please review. Thanks.

File: src/FrontDoor/FrontDoor.Test/ScenarioTests/Common.ps1 [hunk 1]
```diff
@@ -60,4 +60,17 @@ function Assert-Tags($tags1, $tags2)
             throw "Tag content not equal. Key:$key Tags1:" +  $tags1[$key] + "Tags2:" + $tags2[$key]
         }
     }
+}
+
+<#
+.SYNOPSIS
+Sleep in record mode only
+#>
+function SleepInRecordMode ([int]$SleepIntervalInSec)
```
Comment (cormacpayne) [inline src/FrontDoor/FrontDoor.Test/ScenarioTests/Common.ps1]: @pichandwork this function shouldn't be needed -- you can use the existing `Wait-Seconds` function to do the same thing

File: src/FrontDoor/FrontDoor/ChangeLog.md [hunk 2]
```diff
@@ -17,4 +17,8 @@
     * Overview of change #1
         - Additional information about change #1
 -->
-## Upcoming Release
\ No newline at end of file
+## Upcoming Release
+
+## Version 0.7.2
```
Comment (cormacpayne) [inline src/FrontDoor/FrontDoor/ChangeLog.md]: @pichandwork you can remove this header and put all of the text under the `Upcoming Release` header

File: src/FrontDoor/FrontDoor/Cmdlets/GetAzureRmFrontDoorFrontendEndpoint.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
+﻿using System;
```
Comment (cormacpayne) [inline src/FrontDoor/FrontDoor/Cmdlets/GetAzureRmFrontDoorFrontendEndpoint.cs]: @pichandwork please add a license header to this file

File: src/FrontDoor/FrontDoor/Cmdlets/GetAzureRmFrontDoorFrontendEndpoint.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        ///The Front Door object.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ObjectParameterSet, ValueFromPipeline = true, HelpMessage = "The FrontDoor object.")]
+        [ValidateNotNullOrEmpty]
+        public PSFrontDoor FrontDoorObject { get; set; }
+
+        /// <summary>
+        /// Resource Id of the Front Door
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Resource Id of the Front Door")]
```
Comment (cormacpayne) [inline src/FrontDoor/FrontDoor/Cmdlets/GetAzureRmFrontDoorFrontendEndpoint.cs]: @pichandwork this should be the resource id of the frontend endpoint, not the front door. This also shouldn't be in the same parameter set as `-Name`.

File: src/FrontDoor/FrontDoor/help/Disable-AzFrontDoorCustomDomainHttps.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+ResourceState                    : Enabled
+Id                               : /subscriptions/{guid}/resourcegroups/resourcegroup1
+                                   /providers/Microsoft.Network/frontdoors/frontdoor1/frontendendpoints/frontendpointname1-custom-xyz
+Name                             : frontendpointname1-custom-xyz
+Type                             : Microsoft.Network/frontdoors/frontendendpoints
+```
+
+Disable HTTPS for a custom domain with PSFrontendEndpoint object.
+
+### Example 2: Disable HTTPS for a custom domain with ResourceId.
```
Comment (cormacpayne) [inline src/FrontDoor/FrontDoor/help/Disable-AzFrontDoorCustomDomainHttps.md]: @pichandwork nit: Example 3

**Comment by pichandwork** [comment]
@cormacpayne  Incorporated all review comments.
Additionally removed two parameters(protocol and certificatetype) from Enable-AzFrontDoorCustomDomainHttps as we support only default values for these parameters.

**Pull request merged**